zio-flow/shared/src/main/scala/zio/flow/internal/DurableIndexedStore.scala

```diff
@@ -0,0 +1,47 @@
+package zio.flow.internal
+import org.rocksdb.ColumnFamilyHandle
+import zio.rocksdb.service
+import zio.{Chunk, IO, UIO, ZIO}
+import zio.stream.ZStream
+
+import java.io.IOException
+
+object DurableIndexedStore {
+
+  final case class IndexedStoreLive(rocksDB: service.RocksDB) extends IndexedStore {
+
+    private def incrementPosition(topic: String): IO[IOException, Long] = for {
```

Comment:
I think we are going to have a concurrency issue with the position here because two fibers could both read the same position and write to it.

When we start up we could scan each of the topics and get the initial position and then we could use that to construct our own data structure like a `Ref[Map[String, AtomicLong]]` that we could use to ensure we only ever write once to each slot.

Note that this will cause some other potential issues (e.g. a fiber could "reserve" one of the slots and then fail to write to it) that we will have to handle.

Reply:
Got it.

Reply:
I'm thinking of using `service.transactionDB` from `zio-rocksDB` and atomically(`Transaction.atomically`) performing the following operations :
      * increment the `POSITION` value 
      * insert into db (`put`) the new `POSITION` -> `value`

I'm not able to find a way to perform the database `put` operation along with incrementing `POSITION` in `Ref[Map[String, AtomicLong]]` atomically.

Reply:
Good idea!